airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py

```diff
@@ -0,0 +1,73 @@
+import sys
+from typing import Any, Callable, Optional, Union
+
+import dagger
+from click import Context, get_current_context
+from dagger.api.gen import Client, Container
+from pydantic import BaseModel, Field, PrivateAttr
+
+from ..singleton import Singleton
+
+# TODO ben set up for async
+# from asyncclick import Context, get_current_context
+
+
+# this is a bit of a hack to get around how prefect resolves parameters
+# basically without this, prefect will attempt to access the context
+# before we create it in main.py in order to resolve it as a parameter
+# wrapping it in a function like this prevents that from happening
+def get_context() -> Context:
```

Comment:
You can probably get rid of this if you aren't using Prefect. It only matters if you are wrapping function sin Flow calls